cilium-dbg/cmd/bpf_multicast_subscribers.go

```diff
@@ -0,0 +1,294 @@
+		}
+	},
+}
+
+// subscriberData is used to store the subscribers of a multicast group
+type subscriberData struct {
+	groupAddr   netip.Addr
+	subscribers []maps_multicast.SubscriberV4
+}
+
+func outputGroups(groupV4Map maps_multicast.GroupV4Map, groupAddrs []netip.Addr) error {
+	var allGroups []subscriberData
+
+	for _, groupAddr := range groupAddrs {
+		groupData := subscriberData{
+			groupAddr: groupAddr,
+		}
+
+		subscriberMap, err := groupV4Map.Lookup(groupAddr)
+		if err != nil {
+			return fmt.Errorf("failed to lookup multicast group %s: %s", groupAddr, err)
+		}
+
+		subscribers, err := subscriberMap.List()
+		if err != nil {
+			return fmt.Errorf("failed to list multicast subscribers for group %s: %s", groupAddr, err)
+		}
+
+		for _, sub := range subscribers {
+			groupData.subscribers = append(groupData.subscribers, *sub)
+		}
+
+		allGroups = append(allGroups, groupData)
+	}
+
+	if command.OutputOption() {
+		if err := command.PrintOutput(allGroups); err != nil {
+			return fmt.Errorf("error getting output of map in %s: %s", command.OutputOptionString(), err)
+		}
+		return nil
+	}
+
+	printSubscriberList(allGroups)
+	return nil
+}
+
+func printSubscriberList(subscribers []subscriberData) {
+	sort.Slice(subscribers, func(i, j int) bool {
+		return subscribers[i].groupAddr.Compare(subscribers[j].groupAddr) < 0
+	})
+
+	// sort subscribers in each group
+	for _, group := range subscribers {
+		sort.Slice(group.subscribers, func(i, j int) bool {
+			return group.subscribers[i].SAddr.Compare(group.subscribers[j].SAddr) < 0
+		})
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 16, 1, 0, ' ', 0)
+	fmt.Fprintln(w, "Group\tSubscriber\tType\t")
+	for _, subData := range subscribers {
+		fmt.Fprintf(w, "%s\t", subData.groupAddr)
+		for i, sub := range subData.subscribers {
+			if i > 0 {
+				// move by one tab to accommodate group address
+				fmt.Fprintf(w, "\t")
+			}
+			fmt.Fprintf(w, "%s\t", sub.SAddr)
+			fmt.Fprintf(w, "%s\t", getSubscriberType(sub.IsRemote))
+			fmt.Fprintf(w, "\n")
+		}
+
+		if len(subData.subscribers) == 0 {
+			fmt.Fprintf(w, "\n")
+		}
+	}
+	w.Flush()
+}
+
+func getSubscriberType(isRemote bool) string {
+	if isRemote {
+		return remoteKW
+	}
+	return localKW
+}
+
+func parseMulticastGroupSubscriberListArgs(args []string) (netip.Addr, bool, error) {
+	if len(args) != 1 {
+		return netip.Addr{}, false, fmt.Errorf("expected exactly one argument")
+	}
+
+	if args[0] == allGroupsKW {
+		return netip.Addr{}, true, nil
+	}
+
+	addr, err := netip.ParseAddr(args[0])
+	if err != nil {
+		return netip.Addr{}, false, fmt.Errorf("invalid group address: %s", err)
+	}
+
+	return addr, false, nil
+}
+
+func parseMulticastGroupSubscriberArgs(args []string) (groupAddr, subscriberAddr netip.Addr, err error) {
+	if len(args) != 2 {
+		return netip.Addr{}, netip.Addr{}, fmt.Errorf("expected exactly two arguments")
+	}
+
+	groupAddr, err = netip.ParseAddr(args[0])
```

Comment:
This seems different for how we parse the group above with additional check for ipv4 and multicast. 
If these are requirements we should be consistent on how we parse the group args. 
Additionally we should reuse group parsing call here and not bleed out to the call site.